runtime/service/src/main/java/org/apache/polaris/service/quarkus/config/QuarkusProducers.java

```diff
@@ -170,6 +175,34 @@ public UserSecretsManagerFactory userSecretsManagerFactory(
     return userSecretsManagerFactories.select(Identifier.Literal.of(config.type())).get();
   }
 
+  @Produces
+  @Singleton
+  @Identifier("http-client-s3")
+  public SdkHttpClient sdkHttpClient(S3AccessConfig config) {
```

Comment:
if we are having shared connection pool, which will be used by multiple clients which i assume have different routes , wouldnt we want to restrict per route max connection ? to avoid starvation ?